### PR TITLE
Labels functionality and screen

### PR DESCRIPTION
**Pull request by dev-enforced**
In this Pull Request, one can find the following features added to the Unote Notes application. 

1. There are 3 sections related to priority,color palette and label section. Clicking on one closes the other so that the contents of both the sections do not merge on each other avoiding visual confusion at time!!
2. The section related to Labels has 2 labels initially related to School and Office. To add a new label,user can provide input in the mentioned section and press enter
3. The user has the choice to select deselect the labels for his notes
4. The user can even remove the labels from the global list.
5.The user can view the labels after the note gets created.

The files have been formatted using Prettier VScode extension so that's why all files have been formatted or indented in a different manner
Live preview Link: https://notes-app-recreated-git-labels-functionalit-fd0e9e-dev-enforced.vercel.app/

Styling work on Labels as well as note input and card will be done on in the upcoming PRs . Feedback on those areas is really appreciated.

### CODE REVIEW

File: src/components/LabelSection/LabelSection.jsx [hunk 1]
```diff
@@ -1,10 +1,125 @@
...
+        }));
+    };
+
+    const setNoteLabelsOnButtonClick = (e) => {
+        if (e.keyCode === 13) {
+            e.preventDefault();
+            if (labelInput.trim() !== "") {
+                setNoteLabels((prevList) =>
+                    prevList.includes(labelInput)
+                        ? [...prevList]
```
Comment (mariyasada) [inline src/components/LabelSection/LabelSection.jsx]: what is the use of keynode here?

**Review by mariyasada — APPROVED** [review]
liked that you have used reducer for small things also

File: src/components/LabelSection/LabelSection.jsx (hunk 1 above)
Comment (dev-enforced) [reply src/components/LabelSection/LabelSection.jsx]: To make sure that the action of adding a tag happens when we press enter key

File: src/components/LabelSection/LabelSection.jsx [hunk 2]
```diff
@@ -1,10 +1,125 @@
...
+
+
+
+    const [labelInput, setLabelInput] = useState("");
+
+    const removeLabelFromCancelClick = (givenLabel) => {
+        if (noteData.labels.includes(givenLabel)) {
+            dispatchNoteData({
+                type: "REMOVE_LABEL",
+                payload: givenLabel,
```
Comment (tusharkandpal) [inline src/components/LabelSection/LabelSection.jsx]: i will suggest you to give payload as an object of key, value pair.

**Review by tusharkandpal — APPROVED** [review]
very well implemented & minutely went over the details.

File: src/components/LabelSection/LabelSection.css [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    background-color: transparent;
+    cursor: pointer;
+    color: var(--bg-color);
+}
+
+.label-dropdown-section {
+    position: absolute;
+    top: 2.5rem;
+    z-index: 2;
+    background-color: #d3d3d3;
```
Comment (dikshadhote) [inline src/components/LabelSection/LabelSection.css]: use color name

File: src/components/LabelSection/LabelSection.css [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    box-shadow: 1px 2px 10px #cbcaca;
```
Comment (dikshadhote) [inline src/components/LabelSection/LabelSection.css]: same for this

Comment (dev-enforced) [reply src/components/LabelSection/LabelSection.css]: sure will implement it

File: src/components/LabelSection/LabelSection.css (hunk 3 above)
Comment (dev-enforced) [reply src/components/LabelSection/LabelSection.css]: sure will implement it

**Review by dikshadhote — APPROVED** [review]
Overall well  put code.
Just because of the formatting issue it was not readable.

- I will suggest to remove persian blue background color from pallete and use grey as used in label or priority. As this is light theme background takes all attention rather than colors. Actionable items are colors.so, they need attention here.

image

- To display labels chips can be used

**Pull request merged**